code/game/gamemodes/miniantags/guardian/types/assassin.dm

```diff
@@ -1,23 +1,30 @@
 /mob/living/simple_animal/hostile/guardian/assassin
-	melee_damage_lower = 15
-	melee_damage_upper = 15
-	armour_penetration_percentage = 0
-	playstyle_string = "As an <b>Assassin</b> type you do medium damage and have no damage resistance, but can enter stealth, massively increasing the damage of your next attack and causing it to ignore armor. Stealth is broken when you attack or take damage."
+	melee_damage_lower = 20
+	melee_damage_upper = 20
+	damage_transfer = 0.6
+	playstyle_string = "As an <b>Assassin</b> type you do medium damage and have moderate damage resistance, but can enter stealth, massively increasing the damage of your next attack and causing it to ignore armor. Stealth is broken when you attack or take damage."
 	magic_fluff_string = "..And draw the Space Ninja, a lethal, invisible assassin."
 	tech_fluff_string = "Boot sequence complete. Assassin modules loaded. Holoparasite swarm online."
 	bio_fluff_string = "Your scarab swarm finishes mutating and stirs to life, capable of sneaking and stealthy attacks."
+	stealthy_deploying = TRUE
+	attacktext = "slashes"
+	attack_sound = 'sound/weapons/bladeslice.ogg'
 	var/toggle = FALSE
 	var/stealthcooldown = 0
-	var/default_stealth_cooldown = 16 SECONDS
+	var/default_stealth_cooldown = 10 SECONDS
 	var/obj/screen/alert/canstealthalert
 	var/obj/screen/alert/instealthalert
 
-/mob/living/simple_animal/hostile/guardian/assassin/Life(seconds, times_fired)
+/mob/living/simple_animal/hostile/guardian/assassin/Initialize(mapload, mob/living/host)
 	. = ..()
-	updatestealthalert()
+	remove_from_all_data_huds()
 	if(loc == summoner && toggle)
 		ToggleMode(0)
 
+/mob/living/simple_animal/hostile/guardian/assassin/Life(seconds, times_fired)
+	. = ..()
+	updatestealthalert()
```

Comment:
Should this really be done on every single `Life`? Surely it can just be done whenever we actually change stealth modes

Reply:
It also needs to be toggled when the ability goes off cooldown and is usable again, not just when used.

Reply:
Wouldn't it make sense to use a timer then in that case? Start a timer that once the cooldown ends which calls this proc

Reply:
I'd like to see this, but also the proc was already called in `Life`. So technically out of scope.